apps/tangle-website/src/components/leaderboard/RankingTableSection.tsx

```diff
@@ -0,0 +1,155 @@
+import { useState, useEffect } from 'react';
+import cx from 'classnames';
+import {
+  useReactTable,
+  createColumnHelper,
+  getCoreRowModel,
+  getPaginationRowModel,
+  flexRender,
+} from '@tanstack/react-table';
+import { Input, Typography } from '@webb-tools/webb-ui-components';
+import { Search } from '@webb-tools/icons';
+
+import {
+  AddressCell,
+  BadgesCell,
+  HeaderCell,
+  Pagination,
+  RankingItemType,
+} from './tableComponents';
+
+const pageSize = 20;
+
+const columnHelper = createColumnHelper<RankingItemType>();
+
+const columns = [
+  columnHelper.accessor('points', {
+    header: () => <HeaderCell title="Rank" />,
+    cell: (points) => (
+      <Typography variant="mkt-small-caps" fw="bold">
+        #{points.row.index + 1}
+      </Typography>
+    ),
+  }),
+  columnHelper.accessor('address', {
+    header: () => <HeaderCell title="Address" />,
+    cell: (address) => <AddressCell address={address.getValue()} />,
+  }),
+  columnHelper.accessor('badges', {
+    header: () => <HeaderCell title="Badges" />,
+    cell: (badges) => <BadgesCell badges={badges.getValue()} />,
+  }),
+  columnHelper.accessor('points', {
+    header: () => <HeaderCell title="Points" />,
+    cell: (points) => (
+      <Typography variant="mkt-small-caps" fw="bold" ta="center">
+        {points.renderValue()}
+      </Typography>
+    ),
+  }),
+];
+
+export const RankingTableSection = () => {
+  const [rankingData, setRankingData] = useState<RankingItemType[]>([]);
+
+  const {
+    getHeaderGroups,
+    getRowModel,
+    setPageSize,
+    getPageCount,
+    getState,
+    setPageIndex,
+    previousPage,
+    nextPage,
+    getCanPreviousPage,
+    getCanNextPage,
+  } = useReactTable({
+    data: rankingData,
+    columns,
+    getCoreRowModel: getCoreRowModel(),
+    getPaginationRowModel: getPaginationRowModel(),
+    filterFns: {
+      fuzzy: () => {
+        return true;
+      },
+    },
+  });
+
+  const pageIndex = getState().pagination.pageIndex;
+
+  useEffect(() => {
+    setPageSize(pageSize);
+  }, [setPageSize]);
```

Comment:
Why don't we set the `pageSize` in the `useReactTable` options, rather than using `useEffect` and `setPageSize` here? Since we don't provide functionality for users to change the page size, it seems we wouldn't need to utilize the `setPageSize` function.